ui/src/components/AppHeader/AppHeader.vue

```diff
@@ -25,21 +25,80 @@
       <cv-header-global-action :aria-label="$t('ariaUser')">
         <UserAvatar20 />
       </cv-header-global-action>
+      <cv-overflow-menu :flip-menu="ltr" :label="$t('ariaLanguageSetting')">
+        <template slot="trigger">
+          <Language32 />
+        </template>
+        <cv-overflow-menu-item
+          v-for="entry in languages"
+          :key="entry.title"
+          @click="changeLocale(entry.language)"
+        >
+          {{ entry.title }}
+        </cv-overflow-menu-item>
+      </cv-overflow-menu>
     </template>
     <template v-slot:left-panels> </template>
   </cv-header>
 </template>
 
 <script>
 import UserAvatar20 from '@carbon/icons-vue/lib/user--avatar/20';
+import Language32 from '@carbon/icons-vue/lib/language/32';
+import i18n from '@/plugins/i18n';
 
 export default {
   name: 'AppHeader',
-  components: { UserAvatar20 },
+  components: { UserAvatar20, Language32 },
+  mounted() {
+    this.$watch(
+      '$i18n.locale',
+      (newLocale, oldLocale) => {
+        if (newLocale === oldLocale) {
+          return;
+        }
+
+        setDocumentLang(newLocale);
+
+        setDocumentDirectionPerLocale(newLocale);
+      },
+      { immediate: true }
+    );
+  },
   data() {
-    return {};
+    return {
+      ltr: true,
+      languages: [
+        { flag: '', language: 'en', title: 'English' },
+        { flag: '', language: 'es', title: 'Español' },
+        { flag: '', language: 'hi', title: 'हिंदी' },
+        { flag: '', language: 'kr', title: '한국어' },
+        { flag: '', language: 'sc', title: '汉语' },
+        { flag: '', language: 'zh', title: '漢語' },
+        { flag: '', language: 'ru', title: 'русский' },
+        { flag: '', language: 'ar', title: 'اللغة العربية' },
+        { flag: '', language: 'fa', title: ' زبان فارسی' },
+        { flag: '', language: 'ja', title: '日本語' },
+        { flag: '', language: 'tl', title: 'Tagalog' },
+        { flag: '', language: 'vn', title: 'ngôn ngữ tiếng Việt' }
+      ]
+    };
+  },
+  methods: {
+    changeLocale(locale) {
+      i18n.locale = locale;
+      this.ltr = !['ar', 'fa'].includes(locale);
```

Comment:
Added this to resolve issue for ar, fa where the language setting menu moved to the left of the screen and then opened to the left which meant to was clipped by the edge of the browser and very difficult to use. With this change , when the language menu is on the left it opens to the right.

Reply:
ltr menu
![image](https://user-images.githubusercontent.com/7536103/135476159-42937a5c-ddc4-4c23-a750-56ff3dec595d.png)

rtl menu prior to this change. @upkarlidder mentioned this in the chat.
![image](https://user-images.githubusercontent.com/7536103/135476307-0fc72795-b9a6-444d-b0d0-48dd2b0e06c2.png)